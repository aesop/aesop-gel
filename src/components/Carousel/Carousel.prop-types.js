import PropTypes from 'prop-types';

export const IMAGE_PROP_TYPE_SHAPE = PropTypes.object;
export const HYPERLINK_PROP_TYPE_SHAPE = PropTypes.object;

export const SLIDE_PROP_TYPE_SHAPE = PropTypes.exact({
  description: PropTypes.string,
  heading: PropTypes.string,
  image: IMAGE_PROP_TYPE_SHAPE.isRequired,
  url: PropTypes.string,
});

export const INTRODUCTION_PROP_TYPE_SHAPE = PropTypes.exact({
  cta: HYPERLINK_PROP_TYPE_SHAPE,
  description: PropTypes.string.isRequired,
  heading: PropTypes.string.isRequired,
});

export const PROP_TYPES = {
  className: PropTypes.string,
  introduction: INTRODUCTION_PROP_TYPE_SHAPE,
  slides: PropTypes.arrayOf(SLIDE_PROP_TYPE_SHAPE).isRequired,
};

export default PROP_TYPES;
