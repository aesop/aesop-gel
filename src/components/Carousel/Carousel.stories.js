import React from 'react';
import { storiesOf } from '@storybook/react';
import 'normalize.css';
import '~/styles/base.module.css';
import '~.storybook/storybook.module.css';
import CarouselFixture from './Carousel.fixture';
import Carousel from './Carousel';

storiesOf('Carousel', module).add('Base component', () => (
  <Carousel
    introduction={CarouselFixture.introduction}
    slides={CarouselFixture.slides}
  />
));
