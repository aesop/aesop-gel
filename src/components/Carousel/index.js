/* istanbul ignore file */
import { Carousel } from './Carousel';

export { Carousel };
