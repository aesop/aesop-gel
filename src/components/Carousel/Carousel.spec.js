import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import CarouselFixture from './Carousel.fixture';
import Carousel from './Carousel';

configure({ adapter: new Adapter() });

describe('<Carousel />', () => {
  it('should be defined', () => {
    expect(Carousel).toBeDefined();
  });

  it('renders base component correctly', () => {
    const tree = renderer
      .create(
        <Carousel
          introduction={CarouselFixture.introduction}
          slides={CarouselFixture.slides}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders component correctly if no `introduction` prop is provided', () => {
    const tree = renderer
      .create(<Carousel slides={CarouselFixture.slides} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders component correctly if no `url` member is provided on the `slides` prop', () => {
    const tree = renderer
      .create(
        <Carousel
          slides={[
            {
              description: CarouselFixture.slides[0].description,
              heading: CarouselFixture.slides[0].heading,
              image: CarouselFixture.slides[0].image,
            },
          ]}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  // @TODO Address hooks + version of react + testing suite
  // it('should return `null` if the length of the `slides` prop is 0', () => {
  //   const component = shallow(<Carousel slides={[]} />);
  //   expect(component.type()).toEqual(null);
  // });
});
