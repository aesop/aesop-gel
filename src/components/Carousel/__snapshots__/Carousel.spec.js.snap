// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> renders base component correctly 1`] = `
<div
  className="base dark"
>
  <div
    className="slick-slider carousel slick-initialized"
    dir="ltr"
  >
    <button
      className="base inlineStyle dark base hidden"
      disabled={false}
      onClick={null}
      title="next slide"
      type="button"
    >
      <svg
        className="base chevron dark"
        data-ref=""
        focusable="false"
        height={16}
        role="img"
        style={
          Object {
            "height": "16px",
            "width": "16px",
          }
        }
        tabIndex={-1}
        viewBox="0 0 50 50"
        width={16}
      >
        <g>
          <polygon
            className="chevron"
            points="25,31.3 4.2,10.5 0.1,14.6 25,39.5 50,14.6 45.9,10.5"
          />
        </g>
      </svg>
    </button>
    <div
      className="slick-list"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={null}
      onMouseMove={null}
      onMouseUp={[Function]}
      onTouchCancel={null}
      onTouchEnd={[Function]}
      onTouchMove={null}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <div
        className="slick-track"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "WebkitTransform": "translate3d(0px, 0px, 0px)",
            "WebkitTransition": "",
            "msTransform": "translateX(0px)",
            "opacity": 1,
            "transform": "translate3d(0px, 0px, 0px)",
            "transition": "",
          }
        }
      >
        <div
          aria-hidden={false}
          className="slick-slide slick-active slick-current"
          data-index={0}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="base dark"
            >
              <header
                className="base heading"
              >
                <h2
                  className="base noTopMargin mediumWeightFont xXSmall dark eyebrow"
                >
                  Aenean ac tortor consectetur
                </h2>
                <h3
                  className="base noTopMargin serifFont xLarge dark heading"
                >
                  Maecenas elementum
                </h3>
              </header>
              <p
                className="base dark description hasFooter"
              >
                Maecenas elementum, metus id facilisis placerat, turpis diam sodales velit, quis sodales ipsum nunc nec velit. Aenean ac tortor consectetur, volutpat massa sed, ultricies orci.
              </p>
              <footer
                className="ctaWrapper"
              >
                <a
                  className="base hasIcon left dark"
                  href="/"
                  target="_self"
                  title="Aenean ac tortor"
                >
                  Aenean ac tortor
                   
                  <i
                    aria-hidden="true"
                    className="icon"
                  >
                    <svg
                      className="base rightArrow dark"
                      data-ref=""
                      focusable="false"
                      height={10}
                      role="img"
                      style={
                        Object {
                          "height": "10px",
                          "width": "10px",
                        }
                      }
                      tabIndex={-1}
                      viewBox="0 0 50 50"
                      width={10}
                    >
                      <g>
                        <path
                          className="undefined"
                          d="M30.1,5.3L50,25.1L30.1,45h-6.6l18-17.6H0v-4.8h41.5l-18-17.6h6.6V5.3z"
                        />
                      </g>
                    </svg>
                  </i>
                </a>
              </footer>
            </div>
          </div>
        </div>
        <div
          aria-hidden={false}
          className="slick-slide slick-active"
          data-index={1}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="slideWrapper"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <a
                className="base left dark item link"
                href="/"
                target="_self"
                title="Link to Quisque non enim efficitur"
              >
                <div
                  className="information dark"
                >
                  <h5
                    className="base xXSmall dark heading dark"
                  >
                    Quisque non enim efficitur
                  </h5>
                  <div
                    className="description"
                  >
                    Vivamus sollicitudin ac ligula
                  </div>
                </div>
                <div
                  className="imageContainer"
                >
                  <picture
                    className="base picture dark image"
                  >
                    <source
                      media="(min-width: 1025px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <source
                      media="(min-width: 640px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <source
                      media="(min-width: 0px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <img
                      alt="Quisque non enim efficitur"
                    />
                  </picture>
                </div>
              </a>
            </div>
          </div>
        </div>
        <div
          aria-hidden={false}
          className="slick-slide slick-active"
          data-index={2}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="slideWrapper"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <a
                className="base left dark item link"
                href="/"
                target="_self"
                title="Link to Phasellus posuere porta varius"
              >
                <div
                  className="information dark"
                >
                  <h5
                    className="base xXSmall dark heading dark"
                  >
                    Phasellus posuere porta varius
                  </h5>
                  <div
                    className="description"
                  >
                    Suspendisse elit mauris
                  </div>
                </div>
                <div
                  className="imageContainer"
                >
                  <picture
                    className="base picture dark image"
                  >
                    <source
                      media="(min-width: 1025px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <source
                      media="(min-width: 640px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <source
                      media="(min-width: 0px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <img
                      alt="Phasellus posuere porta varius"
                    />
                  </picture>
                </div>
              </a>
            </div>
          </div>
        </div>
        <div
          aria-hidden={false}
          className="slick-slide slick-active"
          data-index={3}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="slideWrapper"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <a
                className="base left dark item link"
                href="/"
                target="_self"
                title="Link to Etiam at pharetra nisl"
              >
                <div
                  className="information dark"
                >
                  <h5
                    className="base xXSmall dark heading dark"
                  >
                    Etiam at pharetra nisl
                  </h5>
                  <div
                    className="description"
                  >
                    Aliquam metus enim, pellentesque eu
                  </div>
                </div>
                <div
                  className="imageContainer"
                >
                  <picture
                    className="base picture dark image"
                  >
                    <source
                      media="(min-width: 1025px)"
                      srcSet="https://www.aesop.com/medias/Aesop-Resurrection-Duet-with-Product-Large-1584x962px.png?context=bWFzdGVyfGltYWdlc3wzOTUwOTh8aW1hZ2UvcG5nfGltYWdlcy9oMWMvaDk4LzkxMTA3OTc2NDc5MDIucG5nfGYwMTE1NGQ3N2M0MDBmYTg1NzY4OTlkNTg0NTlmNmYzMWRkYmJkYzczYjFlOGEyMGMyOWVlZDRmZTRkYzE5ZjI"
                    />
                    <source
                      media="(min-width: 640px)"
                      srcSet="https://www.aesop.com/medias/Aesop-Resurrection-Duet-with-Product-Medium-1238x752px.png?context=bWFzdGVyfGltYWdlc3wyNDczMDF8aW1hZ2UvcG5nfGltYWdlcy9oNjYvaDE5LzkxMTA3OTc4NzcyNzgucG5nfDk4Y2RjZjY4NDk3MGM4YzQ3YzMyMzY4NDhlYjNjNDFhNGNjNGM5MzE1NzFjNDc3YTg3NGIyNDdlYmQ1Yzc1NTA"
                    />
                    <source
                      media="(min-width: 0px)"
                      srcSet="https://www.aesop.com/medias/Aesop-Resurrection-Duet-with-Product-Small-863x524px.png?context=bWFzdGVyfGltYWdlc3wxMjYxNTN8aW1hZ2UvcG5nfGltYWdlcy9oOTgvaDRmLzkxMTA3OTc5MTAwNDYucG5nfDExYTg3YzY4MmZlZTBlODU1NWJiN2IzNzBhNWYwZmExNzg2NGMzNDZhMWQyM2VjZWE4ZWVhMDljMjVmMmRiYzA"
                    />
                    <img
                      alt="Etiam at pharetra nisl"
                    />
                  </picture>
                </div>
              </a>
            </div>
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide"
          data-index={4}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="slideWrapper"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <a
                className="base left dark item link"
                href="/"
                target="_self"
                title="Link to Phasellus posuere porta varius"
              >
                <div
                  className="information dark"
                >
                  <h5
                    className="base xXSmall dark heading dark"
                  >
                    Phasellus posuere porta varius
                  </h5>
                  <div
                    className="description"
                  >
                    Suspendisse elit
                  </div>
                </div>
                <div
                  className="imageContainer"
                >
                  <picture
                    className="base picture dark image"
                  >
                    <source
                      media="(min-width: 1025px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <source
                      media="(min-width: 640px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <source
                      media="(min-width: 0px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <img
                      alt="Phasellus posuere porta varius"
                    />
                  </picture>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="base inlineStyle dark base"
      disabled={false}
      onClick={[Function]}
      title="next slide"
      type="button"
    >
      <svg
        className="base chevron dark"
        data-ref=""
        focusable="false"
        height={16}
        role="img"
        style={
          Object {
            "height": "16px",
            "width": "16px",
          }
        }
        tabIndex={-1}
        viewBox="0 0 50 50"
        width={16}
      >
        <g>
          <polygon
            className="chevron"
            points="25,31.3 4.2,10.5 0.1,14.6 25,39.5 50,14.6 45.9,10.5"
          />
        </g>
      </svg>
    </button>
    <div
      className="base"
    >
      <ul
        className="list dark"
      >
        <li
          className="slick-active"
        >
          <div
            onClick={[Function]}
          />
        </li>
        <li
          className=""
        >
          <div
            onClick={[Function]}
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<Carousel /> renders component correctly if no \`introduction\` prop is provided 1`] = `
<div
  className="base dark"
>
  <div
    className="slick-slider carousel slick-initialized"
  >
    
    <div
      className="slick-list"
    >
      <div
        className="slick-track"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "WebkitTransform": "translate3d(0px, 0px, 0px)",
            "WebkitTransition": "",
            "msTransform": "translateX(0px)",
            "opacity": 1,
            "transform": "translate3d(0px, 0px, 0px)",
            "transition": "",
          }
        }
      >
        <div
          aria-hidden={false}
          className="slick-slide slick-active slick-current"
          data-index={0}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="slideWrapper"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <a
                className="base left dark item link"
                href="/"
                target="_self"
                title="Link to Quisque non enim efficitur"
              >
                <div
                  className="information dark"
                >
                  <h5
                    className="base xXSmall dark heading dark"
                  >
                    Quisque non enim efficitur
                  </h5>
                  <div
                    className="description"
                  >
                    Vivamus sollicitudin ac ligula
                  </div>
                </div>
                <div
                  className="imageContainer"
                >
                  <picture
                    className="base picture dark image"
                  >
                    <source
                      media="(min-width: 1025px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <source
                      media="(min-width: 640px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <source
                      media="(min-width: 0px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <img
                      alt="Quisque non enim efficitur"
                    />
                  </picture>
                </div>
              </a>
            </div>
          </div>
        </div>
        <div
          aria-hidden={false}
          className="slick-slide slick-active"
          data-index={1}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="slideWrapper"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <a
                className="base left dark item link"
                href="/"
                target="_self"
                title="Link to Phasellus posuere porta varius"
              >
                <div
                  className="information dark"
                >
                  <h5
                    className="base xXSmall dark heading dark"
                  >
                    Phasellus posuere porta varius
                  </h5>
                  <div
                    className="description"
                  >
                    Suspendisse elit mauris
                  </div>
                </div>
                <div
                  className="imageContainer"
                >
                  <picture
                    className="base picture dark image"
                  >
                    <source
                      media="(min-width: 1025px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <source
                      media="(min-width: 640px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <source
                      media="(min-width: 0px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <img
                      alt="Phasellus posuere porta varius"
                    />
                  </picture>
                </div>
              </a>
            </div>
          </div>
        </div>
        <div
          aria-hidden={false}
          className="slick-slide slick-active"
          data-index={2}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="slideWrapper"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <a
                className="base left dark item link"
                href="/"
                target="_self"
                title="Link to Etiam at pharetra nisl"
              >
                <div
                  className="information dark"
                >
                  <h5
                    className="base xXSmall dark heading dark"
                  >
                    Etiam at pharetra nisl
                  </h5>
                  <div
                    className="description"
                  >
                    Aliquam metus enim, pellentesque eu
                  </div>
                </div>
                <div
                  className="imageContainer"
                >
                  <picture
                    className="base picture dark image"
                  >
                    <source
                      media="(min-width: 1025px)"
                      srcSet="https://www.aesop.com/medias/Aesop-Resurrection-Duet-with-Product-Large-1584x962px.png?context=bWFzdGVyfGltYWdlc3wzOTUwOTh8aW1hZ2UvcG5nfGltYWdlcy9oMWMvaDk4LzkxMTA3OTc2NDc5MDIucG5nfGYwMTE1NGQ3N2M0MDBmYTg1NzY4OTlkNTg0NTlmNmYzMWRkYmJkYzczYjFlOGEyMGMyOWVlZDRmZTRkYzE5ZjI"
                    />
                    <source
                      media="(min-width: 640px)"
                      srcSet="https://www.aesop.com/medias/Aesop-Resurrection-Duet-with-Product-Medium-1238x752px.png?context=bWFzdGVyfGltYWdlc3wyNDczMDF8aW1hZ2UvcG5nfGltYWdlcy9oNjYvaDE5LzkxMTA3OTc4NzcyNzgucG5nfDk4Y2RjZjY4NDk3MGM4YzQ3YzMyMzY4NDhlYjNjNDFhNGNjNGM5MzE1NzFjNDc3YTg3NGIyNDdlYmQ1Yzc1NTA"
                    />
                    <source
                      media="(min-width: 0px)"
                      srcSet="https://www.aesop.com/medias/Aesop-Resurrection-Duet-with-Product-Small-863x524px.png?context=bWFzdGVyfGltYWdlc3wxMjYxNTN8aW1hZ2UvcG5nfGltYWdlcy9oOTgvaDRmLzkxMTA3OTc5MTAwNDYucG5nfDExYTg3YzY4MmZlZTBlODU1NWJiN2IzNzBhNWYwZmExNzg2NGMzNDZhMWQyM2VjZWE4ZWVhMDljMjVmMmRiYzA"
                    />
                    <img
                      alt="Etiam at pharetra nisl"
                    />
                  </picture>
                </div>
              </a>
            </div>
          </div>
        </div>
        <div
          aria-hidden={false}
          className="slick-slide slick-active"
          data-index={3}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="slideWrapper"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <a
                className="base left dark item link"
                href="/"
                target="_self"
                title="Link to Phasellus posuere porta varius"
              >
                <div
                  className="information dark"
                >
                  <h5
                    className="base xXSmall dark heading dark"
                  >
                    Phasellus posuere porta varius
                  </h5>
                  <div
                    className="description"
                  >
                    Suspendisse elit
                  </div>
                </div>
                <div
                  className="imageContainer"
                >
                  <picture
                    className="base picture dark image"
                  >
                    <source
                      media="(min-width: 1025px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <source
                      media="(min-width: 640px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <source
                      media="(min-width: 0px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <img
                      alt="Phasellus posuere porta varius"
                    />
                  </picture>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    
    
  </div>
</div>
`;

exports[`<Carousel /> renders component correctly if no \`url\` member is provided on the \`slides\` prop 1`] = `
<div
  className="base dark"
>
  <div
    className="slick-slider carousel slick-initialized"
  >
    
    <div
      className="slick-list"
    >
      <div
        className="slick-track"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "WebkitTransform": "translate3d(0px, 0px, 0px)",
            "WebkitTransition": "",
            "msTransform": "translateX(0px)",
            "opacity": 1,
            "transform": "translate3d(0px, 0px, 0px)",
            "transition": "",
          }
        }
      >
        <div
          aria-hidden={false}
          className="slick-slide slick-active slick-current"
          data-index={0}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="slideWrapper"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            >
              <div
                className="item"
              >
                <div
                  className="information dark"
                >
                  <h5
                    className="base xXSmall dark heading dark"
                  >
                    Quisque non enim efficitur
                  </h5>
                  <div
                    className="description"
                  >
                    Vivamus sollicitudin ac ligula
                  </div>
                </div>
                <div
                  className="imageContainer"
                >
                  <picture
                    className="base picture dark image"
                  >
                    <source
                      media="(min-width: 1025px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <source
                      media="(min-width: 640px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <source
                      media="(min-width: 0px)"
                      srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
                    />
                    <img
                      alt="Quisque non enim efficitur"
                    />
                  </picture>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    
    
  </div>
</div>
`;
